### ISSUE TITLE
nbsp HTML entity cannot be put into FAQ title

### ISSUE THREAD

**Issue by MikeMcC399** [corona-warn-app/cwa-website#2163]
## Where to find the issue

cwa-website FAQ titles

## Describe the issue

If the HTML entity `&nbsp;` is put into the title of an FAQ article to insert a non-breaking space it appears literally as `&nbsp;` instead of being interpreted. I had noticed this in the past and recently it had to be corrected by @dsarkar in PR https://github.com/corona-warn-app/cwa-website/pull/2159.

## Steps to reproduce

In [src/data/faq.json](https://github.com/corona-warn-app/cwa-website/blob/master/src/data/faq.json) add a new FAQ article:

```
                    {
                        "title": "Test&nbsp;title",
                        "anchor": "test_title",
                        "active": true,
                        "textblock": [
                            "Test&nbsp;text"
                        ]
                    }
```

Execute `npm run build`

View `public\en\faq\index.html` in web browser and in source code

```
              <div id="test_title-div">
                <dt class="accordion-header active">
                  <h3 class="accordion-item-title" id="test_title">Test&amp;nbsp;title</h3>
                  <span class="accordion-item-icon"></span>
                </dt>
                <dd class="accordion-body">
                  <div class="accordion-item-content">
                    <p>Test&nbsp;text.</p>
                    <p><a href="#test_title" class="faq-anchor">Link to this answer</a></p>
                    <p><a href="#top">To the top</a></p>
                  </div>
                </dd>
              </div>
```

In the title the `&nbsp;` is displayed literally, whereas in the text block it is used as intended and displays a non-breaking space.

![FAQ Test title](https://user-images.githubusercontent.com/66998419/145063727-7cd68785-f3a0-4da3-812a-8707d9cf8cb0.jpg)

## Investigation

The partials [src/partials/page-faq.html](https://github.com/corona-warn-app/cwa-website/blob/master/src/partials/page-faq.html) page, used by Panini to covert the faq.json file to HTML, deals with the title source link differently to the textblock.

https://github.com/corona-warn-app/cwa-website/blob/7a12ee369d8d015579afda5726cc6e40b2ab9e19/src/partials/page-faq.html#L37
https://github.com/corona-warn-app/cwa-website/blob/7a12ee369d8d015579afda5726cc6e40b2ab9e19/src/partials/page-faq.html#L44

Whereas the title uses double curly brackets, the textblock uses triple curly brackets.

I found a reference to this under https://handlebarsjs.com/guide/#html-escaping where is says that triple curly brackets are used to prevent escaping.

## Suggested change

This is not at all my area of expertise, so perhaps somebody with an experienced background in this could take a look.

I did try out changing from double

`<h3 class="accordion-item-title"{{#if anchor}} id="{{anchor}}"{{/if}}>{{title}}</h3>`

to triple curly brackets

`<h3 class="accordion-item-title"{{#if anchor}} id="{{anchor}}"{{/if}}>{{{title}}}</h3>`

I then added back the `&nbsp;` in the title of https://www.coronawarn.app/de/faq/#vac_cert_multiple_scan_possible and it did seem to solve the issue. `&nbsp;` is no longer displayed literally and is instead included in the HTML source code, so it can cause a non-breaking space to be used.

**Comment by MikeMcC399** [comment]
Without the change, reducing the width of the browser window viewing https://www.coronawarn.app/de/faq/#vac_cert_multiple_scan_possible can cause "z. B." to be broken across two lines:

![image](https://user-images.githubusercontent.com/66998419/145065636-b92b7584-6d8e-49fc-95aa-2f9abe2eb4f0.png)

If the partials source is changed, this unwanted break is prevented:

![image](https://user-images.githubusercontent.com/66998419/145066232-9cca03a0-ab5c-4274-b2dc-6e7438c1f7e4.png)

**Comment by dsarkar** [comment]
@MikeMcC399 Thanks! Do you have a PR ready?

**Comment by MikeMcC399** [comment]
@dsarkar 
I have submitted PR #2165 now. It solves the issue and I can't see any unwanted side effects, so it looks like it is right.

I would leave it up to you if you want to actually revert your commit https://github.com/corona-warn-app/cwa-website/commit/7a12ee369d8d015579afda5726cc6e40b2ab9e19 from PR https://github.com/corona-warn-app/cwa-website/pull/2159 to put the `&nbsp;` back into the title of https://www.coronawarn.app/de/faq/#vac_cert_multiple_scan_possible (after merging PR #2165). I tested that you could do this and that it would work.

**Comment by MikeMcC399** [comment]
This issue is resolved by PR #2165.
